hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/ProcessorMetaSupplier.java

```diff
@@ -393,5 +407,35 @@ default boolean snapshottingEnabled() {
          * Returns the guarantee for current job.
          */
         ProcessingGuarantee processingGuarantee();
+
+
+        /**
+         * Returns a temporary directory which contains the attached files to the
+         * job with the given id.
+         *
+         * @param id identifier defined on the {@link ResourceConfig} to be
+         *           used retrieve files from storage.
+         * @return {@link File} handle to a temporary directory which contains the
+         * files attached to the job with provided identifier.
+         */
+        default File getJobFileStorageById(String id) {
```

Comment:
shouldn't this be part of the context ? It seems strange to add it to PMS? Is it meant to be overriden?